protocols/protocol/cloudstate/entity.proto

```diff
@@ -42,22 +61,43 @@ message Forward {
     google.protobuf.Any payload = 3;
 }
 
-// An action for the client
+//ClientAction represents a side effect to be done after this command is handled.
+//
+//---
+//
+//`Specification`:
+//
+//  - An `ClientAction` is considered to be valid iff:
+//    - `action` TODO
+//      - `reply` TODO
+//      - `forward` TODO
+//      - `failure` TODO
 message ClientAction {
+    // TODO
     oneof action {
 
-        // Send a reply
+        // Send a reply.
         Reply reply = 1;
 
-        // Forward to another entity
+        // Forward to another entity.
         Forward forward = 2;
 
         // Send a failure to the client
         Failure failure = 3;
```

Comment:
How differs a `Failure` here and one returned as a `EventSourcedStreamOut.Failure`? What is the state of the event sourced entity instance after such a failure. (See also a related question on [service EventSourced](https://github.com/cloudstateio/cloudstate/pull/119/files/0f528fd47903ac5b564a1f0b5757c6b9be4fa0fc#diff-7a8bc6d812c81b30eb3aa1d95f4c60aaR183))